src/main/java/perfume/webservice/perfume/admin/service/PerfumeAdminService.java

```diff
@@ -47,7 +49,7 @@ public Map<String, Object> savePerfume(PerfumeSaveRequestDtoList requestDto) {
 
```

Comment:
class 자체에 transactional readonly 옵션을 주셨는데, 이유를 여쭤봐도 될까요?

Reply:
JpaRepository를 구현하는 구현체인 
SimpleJpaRepository 를 보시면 
class 전역에  @Transactional(readOnly = true) 선언하고 
데이터 변경이 필요한 메소드에 따로  @Transactional가 선언이 되어있습니다. 
 
```suggestion
@Repository
@Transactional(readOnly = true)
public class SimpleJpaRepository<T, ID> implements JpaRepositoryImplementation<T, ID> {

        @Override
	public long count() {
		return em.createQuery(getCountQueryString(), Long.class).getSingleResult();
	}
 
	@Override
	public long count(@Nullable Specification<T> spec) {
		return executeCountQuery(getCountQuery(spec, getDomainClass()));
	}
 
	@Transactional
	@Override
	public <S extends T> S save(S entity) {
	
}
```

또한 @Transactional(readOnly = true)를 걸어두면  JPA내부적으로 성능 최적화를 한다고 알고 있습니다.

해서 class 전역으로 readonly를 true로 걸어두고 실제 insert나 update 등 데이터 변경이 일어나는곳에서만 따로 @Transactional 을 선언하는 식으로 했습니다.